### ISSUE TITLE
Indexed component not found when adding component through EntityStore.CreateEntity

### ISSUE THREAD

**Issue by dvdking** [friflo/Friflo.Engine.ECS#5]
This test fails
  
```
    [Test]
    public static void Test_Index_ValueStructIndex()
    {
        var store   = new EntityStore();
        store.CreateEntity(new IndexedInt { value = 123 });
        
        var result = store.GetEntitiesWithComponentValue<IndexedInt, int>(123);
        AreEqual(1, result.Count);
    }
```

This does not fail
```
        var store   = new EntityStore();
        var entity1 = store.CreateEntity();
        entity1.AddComponent(new IndexedInt { value = 123 });
        var result = store.GetEntitiesWithComponentValue<IndexedInt, int>(123);
        AreEqual(1, result.Count);
```

batches fail too

```
        var store   = new EntityStore();
        var entity1 = store.CreateEntity();
        
        entity1.Batch().Add(new IndexedInt { value = 123 }).Apply();
        var result = store.GetEntitiesWithComponentValue<IndexedInt, int>(123);
        AreEqual(1, result.Count);
```

**Comment by friflo** [comment]
hey @dvdking 

somebody tries this crazy stuff 👍🏼.
This is currently a known issue for all new [Component Types](https://friflo.gitbook.io/friflo.engine.ecs/examples/component-types).

That is the reason why these new types are only available in 3.0.0-preview releases.
Right now: `AddComponent<>()` and `RemoveComponent<>()` must be used.


I have to make adjustments in all extension methods in [EntityExtensions](https://github.com/friflo/Friflo.Engine-docs/blob/main/api/EntityExtensions.md) and [EntityStoreExtensions](https://github.com/friflo/Friflo.Engine-docs/blob/main/api/EntityStoreExtensions.md) used in your first test.

Same for [EntityBatch](https://github.com/friflo/Friflo.Engine-docs/blob/main/api/EntityBatch.md) and [CreateEntityBatch](https://github.com/friflo/Friflo.Engine-docs/blob/main/api/CreateEntityBatch.md) methods used in your last test.
Without your post I guess I would have missed those two for the final release.

Also the [CommandBuffer](https://github.com/friflo/Friflo.Engine-docs/blob/main/api/CommandBuffer.md) needs adjustments.

thx for feedback!

**Comment by sergiyha** [comment]
Hi. Any news about supporting Indexed Components using CommandBuffers and EntityBatch ?

**Comment by friflo** [comment]
Fixed in 3.0.0-preview.14.

Implemented also all other cases adding / removing / changing indexed components.
Those cases are listed in my initial answer.
